java/gazelle/resolve.go

```diff
@@ -80,25 +80,26 @@ func (Resolver) Embeds(r *rule.Rule, from label.Label) []label.Label {
 
 func (jr Resolver) Resolve(c *config.Config, ix *resolve.RuleIndex, rc *repo.RemoteCache, r *rule.Rule, imports interface{}, from label.Label) {
 	resolveInput := imports.(types.ResolveInput)
-	javaImports := resolveInput.ImportedPackageNames
-	if javaImports.Len() == 0 {
-		return
-	}
 
-	deps := sorted_set.NewSortedSetFn[label.Label]([]label.Label{}, labelLess)
+	isTestRule := isTestRule(r.Kind())
+
+	jr.populateAttr(c, r, "deps", resolveInput.ImportedPackageNames, ix, isTestRule, from, resolveInput.PackageNames)
```

Comment:
What about the `runtime_deps`?

Reply:
Great question! See this comment: https://github.com/bazel-contrib/rules_jvm/blob/9586f4edb573ab9b63569cc1ac6aecc532977bd2/java/gazelle/generate.go#L461-L464

(tl;dr: It's an existing TODO for the future)